ipam/claim.go

```diff
@@ -51,7 +51,15 @@ func (c *claim) Try(alloc *Allocator) bool {
 	if !alloc.ring.Contains(c.cidr.Addr) {
 		// Address not within our universe; assume user knows what they are doing
 		alloc.infof("Address %s claimed by %s - not in our range", c.cidr, c.ident)
-		addOwned()
+		previousOwner := alloc.findOwner(c.cidr.Addr)
+		switch {
+		case previousOwner == "":
+			addOwned()
+		case previousOwner == c.ident: // already owned by this ID
+		case c.ident == api.NoContainerID: // already owned by anonymous container
+		default:
```

Comment:
This line [breaks test 300](https://circleci.com/gh/weaveworks/weave/9189):
```
>>> Test /home/ubuntu/src/github.com/weaveworks/weave/test/300_ip_reuse_test.sh finished after 9.6 secs with error: exit status 1
re-using IP addresses (arp cache updating)
01efe0f974a29b6e09bde505709ea774aa1aebbb9ab10d35b8e3a83a339c765c
71109c15f64b01066b4429767cdb6719efcf96218584331388990a971a52db4e
c2
Unable to claim: address 10.2.1.44/24 already in use
```

The code makes sense to me, but do we expect to be able to transparently re-use IPs like test 300 suggests?
Or should we modify the test to (presumably) either:
- `weave_on $HOST1 detach <cid>`? or
- `proxy docker_on $HOST1 stop <cid>`?